File: lab10/SelectionSort.java
//Simple implementation of Selection Sort in Java

import java.util.Random;
import java.util.Arrays;

public class SelectionSort {

  public static int[] selectionSort(int[] values){
    // Compute length of array
    int size = values.length;

    // TODO: Traverse through all array elements

          // TODO: Find the minimum element in i + 1 onwards

          // TODO: Swap the found min element with ith element

    return values;
  }

  public static void main(String args[]) {
    int size = 10;
    int randArray[] = new int[size];
    Random r = new Random();

    // Populate array with random numbers
    for (int i = 0; i < size; i++) {
      // nextInt(n) returns random number in the range [0,n)
      randArray[i] = r.nextInt(100);
    }

    System.out.println("Array before sorting:");
    // TODO: Print unsorted array (randArray)

    // TODO: Call selection sorting method

    System.out.println("Sorted array:");
    // TODO: Print sorted array returned by selection sort

  } //end main method
} // end class

File: lab10/selectionSort.py
#  Simple implementation of Selection Sort in Python
from random import randint

def selectionSort(values):
    """Takes a list as input, sorts it using
    selection sort, and returns sorted list"""

    # First compute length of list
    size = len(values)

    # Traverse through all list elements
    for i in range(size):

        # Find the index of the minimum element in i + 1 onwards
        minIndex = i
        for j in range(i + 1, size):
            if values[minIndex] > values[j]:
                minIndex = j

        # Swap the found min element with ith element
        values[i], values[minIndex] = values[minIndex], values[i]

    return values

if __name__ == "__main__":
    size = 10
    randList = []

    # Populate list with random numbers
    for _ in range(size):
        randList.append(randint(0, 100))

    # Print unsorted list
    print("List before sorting:")
    print(randList)

    # Call selection sort function
    sortedList = selectionSort(randList)

    # Print sorted list
    print("Sorted list:")
    print(sortedList)
